amf-api-contract/shared/src/main/scala/amf/apicontract/client/platform/AMFConfiguration.scala

```diff
@@ -27,42 +27,106 @@ import amf.core.internal.convert.PayloadValidationPluginConverter.PayloadValidat
+    * @param rl a list of [[ResourceLoader]] to set to the configuration object
+    * @return [[AMFConfiguration]] with [[ResourceLoader]]s set
+    */
   override def withResourceLoaders(rl: ClientList[ResourceLoader]): AMFConfiguration =
     _internal.withResourceLoaders(rl.asInternal.toList)
 
+  /**
+    * Set [[UnitCache]]
+    * @param cache [[UnitCache]] to add to configuration object
+    * @return [[AMFConfiguration]] with [[UnitCache]] added
+    */
   override def withUnitCache(cache: UnitCache): AMFConfiguration =
     _internal.withUnitCache(UnitCacheMatcher.asInternal(cache))
 
+  /**
+    * Add a [[TransformationPipeline]]
+    * @param pipeline [[TransformationPipeline]] to add to configuration object
+    * @return [[AMFConfiguration]] with [[TransformationPipeline]] added
+    */
   override def withTransformationPipeline(pipeline: TransformationPipeline): AMFConfiguration =
     _internal.withTransformationPipeline(pipeline)
 
+  /**
+    * Set [[RenderOptions]]
+    * @param renderOptions [[RenderOptions]] to set to configuration object
+    * @return [[AMFConfiguration]] with [[RenderOptions]] added
+    */
   override def withRenderOptions(renderOptions: RenderOptions): AMFConfiguration =
     _internal.withRenderOptions(renderOptions)
 
+  /**
+    * Set [[ErrorHandlerProvider]]
+    * @param provider [[ErrorHandlerProvider]] to set to configuration object
+    * @return [[AMFConfiguration]] with [[ErrorHandlerProvider]] set
+    */
   override def withErrorHandlerProvider(provider: ErrorHandlerProvider): AMFConfiguration =
     _internal.withErrorHandlerProvider(() => provider.errorHandler())
 
+  /**
+    * Add an [[AMFEventListener]]
+    * @param listener [[AMFEventListener]] to add to configuration object
+    * @return [[AMFConfiguration]] with [[AMFEventListener]] added
+    */
   override def withEventListener(listener: AMFEventListener): AMFConfiguration = _internal.withEventListener(listener)
 
+  /**
+    * Set [[BaseExecutionEnvironment]]
+    * @param executionEnv [[BaseExecutionEnvironment]] to set to configuration object
+    * @return [[AMFConfiguration]] with [[BaseExecutionEnvironment]] set
+    */
   override def withExecutionEnvironment(executionEnv: BaseExecutionEnvironment): AMFConfiguration =
     _internal.withExecutionEnvironment(executionEnv._internal)
 
+  /**
+    * Register a Dialect
+    * @param dialect [[Dialect]] to register
+    * @return [[AMFConfiguration]] with [[Dialect]] registered
+    */
   override def withDialect(dialect: Dialect): AMFConfiguration = _internal.withDialect(asInternal(dialect))
 
+  /**
+    * Register a Dialect
+    * @param path path of the Dialect to register
+    * @return A CompletableFuture of [[AMFConfiguration]]
+    */
   def withDialect(path: String): ClientFuture[AMFConfiguration] = _internal.withDialect(path).asClient
 
   def forInstance(url: String): ClientFuture[AMFConfiguration] = _internal.forInstance(url).asClient
```

Comment:
In forInstance you could add something like:
  /**
    * Register a Dialect linked from a Dialect Instance 
    * @param url path of the Dialect Instance
    * @return A CompletableFuture of [[AMFConfiguration]]
    */